webextensions/browser-compat-data.schema.json

```diff
@@ -0,0 +1,64 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+
+  "definitions": {
+
+    "support_statement": {
+      "type": "object",
+      "properties": {
+        "support": { "type": "string" },
+        "notes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": ["support"],
+      "additionalProperties": false
+    },
+
+    "desktop_browsers": {
+      "type": "object",
+      "properties": {
+        "Chrome": { "$ref": "#/definitions/support_statement" },
+        "Opera": { "$ref": "#/definitions/support_statement" },
+        "Edge": { "$ref": "#/definitions/support_statement" },
+        "Firefox": { "$ref": "#/definitions/support_statement" }
+      },
+      "required": ["Chrome", "Opera", "Edge", "Firefox"],
+      "additionalProperties": false
+    },
+
+    "mobile_browsers": {
+      "type": "object",
+      "properties": {
+        "Firefox OS": { "$ref": "#/definitions/support_statement" },
+        "Edge": { "$ref": "#/definitions/support_statement" },
+        "Firefox": { "$ref": "#/definitions/support_statement" }
+      },
+      "required": ["Firefox OS", "Edge", "Firefox"],
+      "additionalProperties": false
```

Comment:
Sorry, I don't really understand how? Do you mean, because some code that uses this schema might barf if it finds, say, "Safari" in the list?

Do you think it would be better not to use "required" and "additionalProperties"? My reasoning was to catch things like typoes for browser names.